snippets/base/admin.py

```diff
@@ -265,9 +266,62 @@ def preview_url(self, obj):
         return template
     preview_url.allow_tags = True
 
-    def queryset(self, request):
-        return (super(SnippetAdmin, self)
-                .queryset(request).prefetch_related('locales').select_related('template'))
+    def get_queryset(self, request, filtr=True):
+        query = super(SnippetAdmin, self).get_queryset(request)
+        # When filtr it means that this is coming form the SnippetNG admin and
+        # will be filtered later to include only stage page 6 snippets.
+        if filtr:
+            query = query.exclude(on_startpage_6=True)
+        return query.prefetch_related('locales')
```

Comment:
Do we no longer need the `select_related('template')` optimization?

Reply:
This was a optimization put in place back in the day to tackle some SCL3 -> PHX1 connection timeouts. This is not needed anymore and either way it wasn't used b/c `queryset` has been replaced by `get_queryset` a couple Django versions back.

Good point though, thanks!